pkg/apis/v1/ec2nodeclass.go

```diff
@@ -419,6 +423,12 @@ type EC2NodeClass struct {
 
 	// +kubebuilder:validation:XValidation:message="must specify exactly one of ['role', 'instanceProfile']",rule="(has(self.role) && !has(self.instanceProfile)) || (!has(self.role) && has(self.instanceProfile))"
 	// +kubebuilder:validation:XValidation:message="changing from 'instanceProfile' to 'role' is not supported. You must delete and recreate this node class if you want to change this.",rule="(has(oldSelf.role) && has(self.role)) || (has(oldSelf.instanceProfile) && has(self.instanceProfile))"
+	// +kubebuilder:validation:XValidation:message="must set amiFamily to 'AL2' or 'Custom' when using an AL2 alias",rule="!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find('^[^@]+') == 'al2') ? (self.amiFamily == 'Custom' || self.amiFamily == 'AL2') : true)"
+	// +kubebuilder:validation:XValidation:message="must set amiFamily to 'AL2023' or 'Custom' when using an AL2023 alias",rule="!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find('^[^@]+') == 'al2023') ? (self.amiFamily == 'Custom' || self.amiFamily == 'AL2023') : true)"
+	// +kubebuilder:validation:XValidation:message="must set amiFamily to 'Bottlerocket' or 'Custom' when using a Bottlerocket alias",rule="!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find('^[^@]+') == 'bottlerocket') ? (self.amiFamily == 'Custom' || self.amiFamily == 'Bottlerocket') : true)"
+	// +kubebuilder:validation:XValidation:message="must set amiFamily to 'Windows2019' or 'Custom' when using a Windows2019 alias",rule="!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find('^[^@]+') == 'windows2019') ? (self.amiFamily == 'Custom' || self.amiFamily == 'Windows2019') : true)"
+	// +kubebuilder:validation:XValidation:message="must set amiFamily to 'Windows2022' or 'Custom' when using a Windows2022 alias",rule="!has(self.amiFamily) || (self.amiSelectorTerms.exists(x, has(x.alias) && x.alias.find('^[^@]+') == 'windows2022') ? (self.amiFamily == 'Custom' || self.amiFamily == 'Windows2022') : true)"
```

Comment:
Had to replace a single validation rule with a validation rule for each family to stay within cost estimate requirements for k8s <= 1.27. I believe this is due to a cost estimate bug that existed on those versions which overestimated the cost for functions which return strings. `self.amiFamily.lowerAscii()` appears to have been the culprit.

This was the original CEL expression:
```cel
(has(self.amiFamily) && self.amiSelectorTerms.exists(x, has(x.alias))) ? (self.amiFamily == 'Custom' || self.amiSelectorTerms.exists(x, has(x.alias) && (x.alias.find('^[^@]+') == self.amiFamily.lowerAscii()))) : true
```